Dockerfile

```diff
@@ -54,6 +55,13 @@ RUN --mount=type=cache,target=/root/.cache/go-build \
     go build -trimpath -ldflags "${ldflags} -extldflags '-static'" \
     -o manager ${package}
 
+# Build the clusterctl binary. The output binary will be 'clusterctl' rather than the default 'manager'.
+RUN --mount=type=cache,target=/root/.cache/go-build \
+    --mount=type=cache,target=/go/pkg/mod \
+    CGO_ENABLED=0 GOOS=linux GOARCH=${ARCH} \
+    go build -trimpath -ldflags "${ldflags} -extldflags '-static'" \
+    -o clusterctl ${package}
+
```

Comment:
```suggestion
```

I dont think you need a second `RUN` section here. This will lead to building both the manager and the clusterctl binaries (all though one will fail). You can re-purpose the existing `RUN` command so that the same command can be used to build the manager and clusterctl.  Changing the existing `RUN` to something like this should be enough: 
```
RUN --mount=type=cache,target=/root/.cache/go-build \
    --mount=type=cache,target=/go/pkg/mod \
    CGO_ENABLED=0 GOOS=linux GOARCH=${ARCH} \
    go build -trimpath -ldflags "${ldflags} -extldflags '-static'" \
    -o ${outbin} ${package}
```

Note that you will also have to replace "manager" in l.68 and l.71 with `${outbin}`.

Reply:
Thanks. Made the required changes.